src/internal/m365/collection/drive/collections.go

```diff
@@ -990,6 +1161,22 @@ func (c *Collections) processItem(
 			return clues.New("item seen before parent folder").WithClues(ictx)
 		}
 
+		// Don't move items if the new collection's already reached it's limit. This
+		// helps ensure we don't get some pathological case where we end up dropping
+		// a bunch of items that got moved.
+		//
+		// We need to check if the collection already contains the item though since
+		// it could be an item update instead of a move.
+		if !collection.ContainsItem(item) &&
+			limiter.atContainerItemsLimit(collection.AddedItems()) {
+			return nil
+		}
+
+		// Skip large files that don't fit within the size limit.
+		if limiter.enabled() &&
```

Comment:
imo all `limiter.enabled() && limiter.foo()` checks could pack the enabled() check into foo().